core/src/main/java/org/elasticsearch/index/mapper/SourceToParse.java

```diff
@@ -91,6 +97,10 @@ public String routing() {
         return this.routing;
     }
 
+    public XContentType getXContentType() {
+        return this.xContentType;
```

Comment:
I had a look at whether we retrieve the content type whenever we retrieve the source and found that ParsedDocument should probably hold the content type too. Otherwise we end up with auto-detection whenever we retrieve the source from it.